src/custom-element.js

```diff
@@ -123,48 +123,55 @@ export function stubElements(win) {
  * @param {!AmpElement} element
  */
 export function applyLayout_(element) {
+  const layoutAttr = element.getAttribute('layout');
   let widthAttr = element.getAttribute('width');
   let heightAttr = element.getAttribute('height');
   const sizesAttr = element.getAttribute('sizes');
-  const layoutAttr = element.getAttribute('layout');
+  let layout = null;
+  let width = null;
+  let height = null;
+
+  // Layout specified directly.
+  if (layoutAttr) {
+    layout = parseLayout(layoutAttr.trim());
+    if (!layout) {
+      throw new Error('Unknown layout: ' + layoutAttr);
+    }
+  }
 
-  // Handle elements that do not specify a width/height and are defined to have
-  // natural browser dimensions.
-  if ((!layoutAttr || layoutAttr == Layout.FIXED ||
-          layoutAttr == Layout.FIXED_HEIGHT) &&
+  // Default width and height: handle elements that do not specify a
+  // width/height and are defined to have natural browser dimensions.
+  if ((!layout || layout == Layout.FIXED || layout == Layout.FIXED_HEIGHT) &&
       (!widthAttr || !heightAttr) && hasNaturalDimensions(element.tagName)) {
     const dimensions = getNaturalDimensions(element.tagName);
-    if (layoutAttr != Layout.FIXED_HEIGHT) {
+    if (layout != Layout.FIXED_HEIGHT) {
       widthAttr = widthAttr || dimensions.width;
     }
     heightAttr = heightAttr || dimensions.height;
   }
 
-  let layout;
-  if (layoutAttr) {
-    // TODO(dvoytenko): show error state visually in the dev mode, e.g.
-    // red background + error message.
-    layout = parseLayout(layoutAttr.trim());
-    if (!layout) {
-      throw new Error('Unknown layout: ' + layoutAttr);
-    }
-  } else if (widthAttr || heightAttr) {
-    if (!widthAttr || widthAttr == 'auto') {
-      layout = Layout.FIXED_HEIGHT;
+  // Default layout value.
+  if (!layout) {
+    if (widthAttr || heightAttr) {
+      if (!widthAttr || widthAttr == 'auto') {
+        layout = Layout.FIXED_HEIGHT;
```

Comment:
There can't be width:auto - that wouldn't be the natural dimension. You can simply default them to '1' if it's convenient for you code - you don't need to resolve them.